### PR TITLE
UHM-3667 : Fetching Lab orders by date range

### PR DESCRIPTION
**Pull request by rhenshaw56**

### CODE REVIEW

File: app/js/reducers/initialState.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ export default {
     orders: [],
     labTests: [],
     isLoading: false,
+    fetched: false,
```
Comment (jomadoye) [inline app/js/reducers/initialState.js]: What is the difference between `fetched` and `isLoading`, I assume if a data is still being fetched then it is still loading.

File: app/js/utils/helpers.js [hunk 2]
```diff
@@ -45,10 +45,10 @@ export const filterThrough = (filters, data) => {
     originalData = filteredData;
   }
 
-  if (filters.dateToField && filters.dateFromField) {
-    const filteredData = getDateRange(originalData, filters.dateFromField, filters.dateToField, filters.dateField);
-    originalData = filteredData;
-  }
+  // if (filters.dateToField && filters.dateFromField) {
+  //   const filteredData = getDateRange(originalData, filters.dateFromField, filters.dateToField, filters.dateField);
```
Comment (jomadoye) [inline app/js/utils/helpers.js]: Please remove commented code.

File: app/js/components/LabOrdersList.jsx [hunk 3]
```diff
@@ -181,37 +212,34 @@ export class LabOrdersList extends PureComponent {
...
+            clearNameEMRField={this.clearNameEMRField}
+            labTests={labTests}
+            testTypeField={testTypeField}
+            dateFromField={moment(dateFromField)}
+            dateToField={moment(dateToField)}
+            nameField={nameField}
+          />
+          {!fetched && <Loader />}
+          {hasData && this.renderDraftOrderTable()}
+          {(!hasData && fetched) && this.renderNoDataDisplayText()}
```
Comment (jomadoye) [inline app/js/components/LabOrdersList.jsx]: Nice!

File: app/js/reducers/initialState.js (hunk 1 above)
Comment (rhenshaw56) [reply app/js/reducers/initialState.js]: yeah @jomadoye so the isLoading will track the status of fetching just the orders and the fetched will track the status of completion for the fetch calls for that particular view

**Comment by rhenshaw56** [comment]
@jomadoye I agree we would need to do the same for the labResultsList page but this PR is just for the orders, I think the results approach would be more different

**Pull request merged**